Odata-docs/client/bulk-update-operations.md

```diff
@@ -0,0 +1,245 @@
+context.AddRelatedObject(customer, "Orders", order1);
+
+DataServiceResponse response = await context.BulkUpdateAsync(customer);
+```
+
+### Sample Request 3
+This sample request demonstrates how to create a new customer and update the amounts of 2 existing orders then link the updated orders to the created customer.
+
+```http
+PATCH http://localhost:6285/odata/Customers
+```
+Request Body
+```json
+{
+	"@context": "http://localhost:6285/odata/Customers/$delta",
+	"value": [{
+		"Id": 1,
+		"Name": "Customer1",
+		"Orders@delta": [
+            {
+				"@id":"Orders(1)", 
+                "Amount": 14.12
+			},
+			{
+				"@Id": "Orders(2)",
+				"Amount": 23.21
+			}
+		]
+	}]
+}
+```
+
+How to make the above request from OData client using OData Connected Service. 
+
+```csharp
+var context = new Container(new Uri("http://localhost:6285/odata/"));
+
+var customer = new Customer
+{
+    Id = 1,
+    Name = "Customer1",
+    Age = 26
+};
+
+context.AddToCustomers(customer);
+
+DataServiceCollection<Order> ordersCollection = new DataServiceCollection<Order>(context);
+ordersCollection.Load(context.Orders);
+
+ordersCollection[0].Amount = 14.12;
+ordersCollection[1].Amount = 23.21;
+
+context.AddLink(customer, "Orders", ordersCollection[0]);
+context.AddLink(customer, "Orders", ordersCollection[1]);
+
+DataServiceResponse response = await context.BulkUpdateAsync(customer);
+```
+### Sample Request 4
+This sample request demonstrates how to update the age of an existing customer and remove a link to an existing order linked to the customer. 
+
+```http
+PATCH http://localhost:6285/odata/Customers
+```
+Request Body
+```json
+{
+	"@context": "http://localhost:6285/odata/Customers/$delta",
+	"value": [{
+		"Id": 1,
+		"Name": "Customer1",
+		"Age": 56,
+		"Orders@delta": [{
+			"@removed": {
+				"reason": "changed"
+			},
+			"@id": "http://localhost:6285/odata/Orders(2)"
+		}]
+	}]
+}
+```
+How to make the above request from OData client using OData Connected Service. 
+
+```csharp
+var context = new Container(new Uri("http://localhost:6285/odata/"));
+
+var customer = context.Customers.FirstOrDefault(a=>a.Id == 1);
+
+var order = context.Orders.FirstOrDefault(a=>a.Id == 2);
+
+context.DeleteLink(customer, "Orders", order);
+
+DataServiceResponse response = await context.BulkUpdateAsync(customer);
+```
+
+## Things to Note: 
+When calling the `BulkUpdateAsync` method, you need to pass in the top-level objects that you want to perform a bulk update on. 
+
+If the request gets processed successfully, the `DataServiceResponse` returned from the call to `BulkUpdateAsync` will have all the `ChangeOperationResponse`'s for all the operations in the bulk update request. If any of the operations failed, then the failed operation's `ChangeOperationResponse`'s Descriptor's `SaveResultWasProcessed` value will be `Unchanged`. Also, the Descriptor's `SaveError` property will have the error of the failed operation. 
```

Comment:
Maybe you could simulate such a failure and attach a screenshot of Visual Studio with a view of the error expanded.